packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/io.dart';
+import '../doctor_validator.dart';
+
+class WindowsVersionValidator extends DoctorValidator {
+  /// Validator to be run with `flutter doctor` to check
+  /// Windows host machines if they are running supported versions,
+  /// current unsupported versions = 7, 8
+  const WindowsVersionValidator({required ProcessManager processManager})
+      : _processManager = processManager,
+        super('Windows Version');
+
+  final ProcessManager _processManager;
+
+  @override
+  Future<ValidationResult> validate() async {
+    final ProcessResult result;
+    try {
+      result = _processManager.runSync(<String>['systeminfo']);
+    } on ProcessException {
+      return const ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[],
+        statusInfo:
+            'Unable to run Windows version check using built-in `systeminfo`',
```

Comment:
Flutter developers are unlikely to know what this means. I would loop in @christopherfujino and @Jasguerrero , but I would epect that if this check fails because we can't determine the version, we should just not say anything at all rather than worry people.

Reply:
Agreed, I wasn't too sure how to handle errors/failures for the validator... i just added them to handle all errors during testing

Reply:
I think for both this and exitCode != 0, we have two options:

1) show what happened to the user and ask them to file a bug because something we weren't expecting happened; or
2) do what Jonah suggested, and not say anything and not worry people (anytime there's a non-green doctor validator it causes a subset of our users a lot of anxiety)

If you want to do 1, you can just not catch the ProcessException and then explicitly throw an exception for the exitCode != 0 case; uncaught exceptions from doctor validators will be caught higher up and surfaced to the user.